addons/web/static/src/legacy/js/core/utils.js

```diff
@@ -403,8 +403,8 @@ const utils = {
      *  empty or if an error occurs.
      */
     getDataURLFromFile: function (file) {
-        if (!file) {
-            return Promise.reject();
+        if (!file || !file.size) {
+            return Promise.reject(_t("Empty File"));
```

Comment:
You need to add this in the `.pot` file, see https://github.com/odoo/odoo/pull/113570#pullrequestreview-1313445887
Note: need to be inserted exactly where it would be through the export mechanism, so alphabetically to make it easier for futur commit but also forward port and to reduce file diff when comparing the codebase file to the exported one.

Reply:
I caught a slight issue with my changes. As we are going to reject promise when the file will be empty it will cause issues at all those places where `getDataURLFromFile` method is not placed inside a `try/catch`.

Ex. https://github.com/odoo-dev/odoo/blob/16.0/addons/web/static/src/views/fields/file_handler.js#L46 , etc.

So, should we make changes over there as well? 
And should we throw an Error with a proper error type or something like that for empty file instead of a Rejected Promise for better error handling at the caller function?

Reply:
cc @odoo/rd-framework-js

> I caught a slight issue with my changes

Indeed, this change (add `|| !file.size`) is probably not ok in stable.
I noticed during my first review that changing a product image in the backend was crashing in 16.0 but wasn't sure if the framework-js was supposed to handle those cases correctly.
Code you linked seems to be handling this use case correctly but it's not the case everywhere.

I think it's a bit annoying that this util is not handling this use case correctly, but having a stable friendly fix is the most important think so I guess there is not much choice here, we need to handle this case at the caller level (as in the link you provided) and not in the callee/util.
And in master forward port, make the callee/util change so it handle empty file too?

Reply:
It appears that this is a Chrome bug, the [dataURL RFC](https://www.ietf.org/rfc/rfc2397.txt) specifies that the comma is mandatory, but Chrome simply returns `data:` for empty file which is not a valid data URL. Firefox returns ` data:<mimetype>;base64,` (I have not tested safari). Later, the code tries to split on the comma and uses `splitString[1]` which should always be defined even if empty but in this case it's undefined leading to the python crash. In Firefox, you just get a error toast: `Uploaded image's format is not supported. Try with: .gif, .jpe, .jpeg, .jpg, .png, .svg` which seems reasonable. Empty files are still files

In any case, the caller is using this method to obtain a base64 representation of the file data, but whether the dataURL encodes the data as base64 or not is actually [unspecified](https://w3c.github.io/FileAPI/#packaging-data). In practice it is the case but relying on unspecified behaviour doesn't seem like the best idea unless there are no alternatives. It seems we're doing this in a bunch of places however so I guess changing this is out of scope.

I think here the simplest solution would be to account for the bug and construct the dataURL by hand when it's just `data:`, eg:
```js
// Handle Chrome bug that creates invalid data URLs for empty files
if (reader.result === "data:") {
  return `data:${file.type};base64,`
}
return reader.result;
```
We might want to file a bug with the chromium project, I haven't found any referencing this issue.